### DISCUSSION TITLE
Feature: unify the implementation of charge mixing in SCF and mixing used in EXX

### DISCUSSION THREAD

**Discussion by WHUweiqingzhou** [deepmodeling/abacus-develop#7083] [General]
### Background

Currently, the mixing used in EXX is independent with the charge mixing in SCF. Maybe a unified mixing is helpful for better convergence of HSE calculations.

### Describe the solution you'd like

Change mixing implementation to cooperate with the charge mixing implementation in SCF.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by mohanchen** [comment]
That's a great idea! We will try to implement the new function.

**Comment by mohanchen** [comment]
I will transfer this to discussion. Hopefully we will come up with a better solution in near future.
